src/openforms/js/components/formio_builder/builder.js

```diff
@@ -17,6 +18,21 @@ const getBuilderOptions = () => {
   const maxFileUploadSize = jsonScriptToVar('setting-MAX_FILE_UPLOAD_SIZE');
   const formFieldsRequiredDefault = jsonScriptToVar('config-REQUIRED_DEFAULT');
 
+  /**
+   * For use in a FormIO custom validator.
+   *
+   *
+   * @param {object | array | number | string} logic - JsonLogic expression
+   * @param {object | array | number | string} expected - example expected result value [["label", "value"]]
+   * @returns {boolean | string} - OK or an error message
+   */
+  const validateLogic = (logic, expected) => {
+    // We don't evaluate logic, we just look at types.
+    // "===" forces the logic expression align with the expectancy
+    const result = infer({'===': [logic, expected]}, {});
+    return result.startsWith('result type:') || result;
```

Comment:
`infer` API must change. This is ugly. Want to have more info from other tickets before making a decision and releasing a new npm package.